rest-request.go

```diff
@@ -37,10 +38,29 @@ var DefaultHTTPClient = http.DefaultClient
 
 // Client uses Grafana REST API for interacting with Grafana server.
 type Client struct {
-	baseURL   string
-	key       string
-	basicAuth bool
-	client    *http.Client
+	baseURL       string
+	key           string
+	basicAuth     bool
+	client        *http.Client
+	customHeaders map[string]string
+}
+
+// SetCustomHeaders - set additional headers that will be sent with each request
+func (r *Client) SetCustomHeaders(headers map[string]string) {
+	r.customHeaders = headers
+}
+
+// SetCustomHeader - set additional header that will be sent with each request
+func (r *Client) SetCustomHeader(key, value string) {
```

Comment:
I feel like it will be unnecessary confusion for users to have two different APIs for the same purpose. Could we please remove this one as the other one suffices and is clearer i.e. the behaviour is very clear?

Reply:
so, do you propose to leave `SetCustomHeaders` and remove `SetCustomHeader`? I think `SetCustomHeader` easier to use and will be mostly used, but `SetCustomHeaders` still useful if you want to rewrite and clear all custom headers.
WDYT?

Reply:
I think that the typical way in Go to pass key/value pairs is to use `map[KeyType]ValueType` and it has quite a nice syntax. So, I personally don't see a reason to add some extra methods on top of that which simply do `someMap[k] = v`. If someone really wants this functionality then it probably won't be a huge inconvenience to store those headers somewhere on the user side :thinking: we could probably add a custom convenience function - `GetCustomHeaders`. 

I really don't see a point in hiding from the users that it's a `map[string]string` underneath as that is what headers are, really :smile: